src/main/java/com/example/cult_of_tim/cultoftim/repositories/PromotionRepository.java

```diff
@@ -10,15 +10,15 @@
 @Repository
 public interface PromotionRepository extends JpaRepository<Promotion, Long> {
 
-    Optional<Promotion> findById(Long id);
+    //Optional<Promotion> findById(Long id);
 
-    List<Promotion> findAll();
+    //List<Promotion> findAll();
 
     List<Promotion> findByUserId(Long userId);
 
-    boolean existsByIdAndGlobal(Long promotionId, boolean global);
+    boolean existsByIdAndGlobalPromotion(Long promotionId, boolean global);
```

Comment:
I don't know why this is needed, if you need to check if item exists, then just do boolean existsById, if you need to check if this id has global true,
boolean existsByIdAndGlobalPromotionTrue(Long promotionId);
https://www.baeldung.com/spring-data-derived-queries